http.go

```diff
@@ -5,11 +5,52 @@ import (
 	"fmt"
 	"net"
 	"net/http"
-	"net/url"
 	"runtime/debug"
 	"strings"
 )
 
+const (
+	redaction = "********"
+)
+
+// Query fields whose value will redacted. Used by NewHttp.
+var QuerySecretFields = []string{"password", "passphrase", "passwd", "secret"}
+
+// Header fields whose value will redacted. Used by NewHttp.
+var HeaderSecretFields = []string{"Authorization"}
+
+func redactQuery(r *http.Request) string {
+	query := r.URL.Query()
+
+	for _, keyword := range QuerySecretFields {
+		for field := range query {
+			if field == keyword {
+				query[field] = []string{redaction}
+				break
+			}
+		}
+	}
+
+	return query.Encode()
+}
+
+func redactHeaders(r *http.Request) map[string]string {
+	headers := make(map[string]string, len(r.Header))
+
+	for k, v := range r.Header {
+		for _, field := range HeaderSecretFields {
+			if field == k {
+				rep := strings.Repeat(redaction+",", len(v))
```

Comment:
Oh, nevermind, we were turning it into a `map[string]string` before.

I guess the function name is then misleading since it also coerces the headers into a different type.